File: algorithms/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        length = len(nums)
        rec = nums[0]
        i = 1
        while i < length:
            if nums[i - 1] > 0:
                nums[i] = nums[i] + nums[i - 1]
            if rec < nums[i]:
                rec = nums[i]
            i += 1
        return rec

File: algorithms/0053-maximum-subarray.go
package algorithms

func maxSubArray(nums []int) int {
	max := nums[0]
	for i := 1; i < len(nums); i++ {
		if 0 < nums[i - 1] {
			nums[i] = nums[i - 1] + nums[i]
		}
		if max < nums[i] {
			max = nums[i]
		}
	}
	return max
}
